VideoLocker/src/test/java/org/edx/mobile/view/BaseFragmentActivityTest.java

```diff
@@ -302,6 +260,40 @@ public void run() {
     }
 
     /**
+     * Testing show banner method
+     */
+    @Test
+    public void showBannerTest() {
+        final BaseFragmentActivity activity =
+                Robolectric.buildActivity(getActivityClass())
+                        .withIntent(getIntent()).setup().get();
+        final Window window = activity.getWindow();
+        final View banner = window.findViewById(R.id.banner);
+        assertNotNull(banner);
+        assertThat(banner).hasVisibility(View.GONE);
+        final TextView bannerTextView = (TextView) window.findViewById(R.id.banner_text);
+        assertNotNull(bannerTextView);
+        final TextView bannerExpanded = (TextView) window.findViewById(R.id.banner_expanded);
+        assertNotNull(bannerExpanded);
+        assertThat(bannerExpanded).hasVisibility(View.GONE);
+        for (final BannerType bannerType : BannerType.values()) {
+            activity.showBanner(bannerType);
```

Comment:
Instead of showing banner ourselves here, we should instead send the appropriate event on the event bus and then apply the assertions.